superset/sqllab/command.py

```diff
@@ -110,7 +114,13 @@ def run(  # pylint: disable=too-many-statements,useless-suppression
                 "status": status,
                 "payload": self._execution_context_convertor.serialize_payload(),
             }
-        except (SqlLabException, SupersetErrorsException) as ex:
+        except SupersetErrorsException as ex:
+            if all(ex.error_type == SupersetErrorType.SYNTAX_ERROR for ex in ex.errors):
+                ex.status = 422
```

Comment:
This is great @diegomedina248. Thank you! Can we raise a new error instead of the original one? We may have to create a new syntax error class with a 422 status.